microservice_a/build.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+set -xe
+
+__SVC=$( echo "$PWD" | awk -F '/' '{print $NF}')
+
+# just checking null
+[[ -z ${PWD} ]] \
+  || [[ -z $1 ]] \
+  || [[ -z $IMAGE_PREFIX ]] \
+  && exit 1
+
+echo "building $__SVC with tag $1"
+
+# my docker and docker on github actions runner does not work with --link option
+# but I still want to use symlinks
+# so, I will just leave those files symlinks and add making hardlinks from symlinks
+# when running pipeline
+
+find ./ -type l -exec sh -c 'ln -fn $(readlink "{}") "{}"' \;
+
+docker build \
```

Comment:
• Each microservice should have a script (bash, etc.) that builds its docker image.